src/pyclaw/fileio/ascii.py

```diff
@@ -329,39 +329,51 @@ def read_array(f, state, num_var):
     q = np.zeros(q_shape)
 
 
-    if patch.num_dim == 1:
-        for i in range(patch.dimensions[0].num_cells):
-            l = []
-            while len(l)<num_var:
-                line = f.readline()
-                l = l + line.split()
-            for m in range(num_var):
-                q[m,i] = float(l[m])
-    elif patch.num_dim == 2:
-        for j in range(patch.dimensions[1].num_cells):
+    try:
+        if patch.num_dim == 1:
             for i in range(patch.dimensions[0].num_cells):
                 l = []
                 while len(l)<num_var:
                     line = f.readline()
+                    if line=='':
+                        raise IOError('Unexpected EOF in %s' % f.name)
```

Comment:
This seems a bit confusing, at least on the face of it.  If there is just an empty line somewhere in the file, this will claim (to the user) that it's the end of the file.

Maybe say "Unexpected blank line or EOF"?

Reply:
An empty line returns `'\n'` so I think the empty string does indicate the EOF in general.